From 6d736b58defa383916d85629fc787427e8da722f Mon Sep 17 00:00:00 2001
From: Petr Lautrbach <plautrba@redhat.com>
Date: Mon, 26 Oct 2015 18:36:44 +0100
Subject: [PATCH] semanage: Use OrderedDict for list of fcontexts

semanage-fcontext shows which file contexts are defined and
do a sort which does not follow real order. It could be confusing for
users therefore we need to use OrderedDict and preserve the order.

Resolves: rhbz#1206767
---
 semanage/seobject.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/semanage/seobject.py b/semanage/seobject.py
index 0e1ef97..cf018d9 100644
--- a/semanage/seobject.py
+++ b/semanage/seobject.py
@@ -1976,6 +1976,9 @@ class fcontextRecords(semanageRecords):
                        self.flist += fclocal
 
                 ddict = {}
+                from collections import OrderedDict
+                ddict = OrderedDict()
+
 		for fcontext in self.flist:
 			expr = semanage_fcontext_get_expr(fcontext)
 			ftype = semanage_fcontext_get_type(fcontext)
@@ -1992,7 +1995,6 @@ class fcontextRecords(semanageRecords):
                l = []
                fcon_dict = self.get_all(True)
                keys = fcon_dict.keys()
-               keys.sort()
                for k in keys:
                       if fcon_dict[k]:
                              l.append("-a -f '%s' -t %s '%s'" % (k[1], fcon_dict[k][2], k[0]))
@@ -2002,7 +2004,6 @@ class fcontextRecords(semanageRecords):
 		fcon_dict = self.get_all(locallist)
                 keys = fcon_dict.keys()
 		if len(keys) != 0:
-			keys.sort()
 			if heading:
 				print "%-50s %-18s %s\n" % (_("SELinux fcontext"), _("type"), _("Context"))
 			for k in keys:
-- 
2.6.1

